doc/source/installation.rst

```diff
@@ -1,140 +1,215 @@
+
+- ``analysis``: optional dependencies for various analysis modules.
+- ``extra_formats``: support for additional file formats.
+- ``parallel``: install dask to enable parallel analysis across multiple cores or machines.
+- ``doc``: install documentation build dependencies (e.g. Sphinx, themes).
+
+To see the full list of available extras, check the `pyproject.toml file`_.
+
+To install with optional features, use pip with one or more extras:
 
 .. code-block:: bash
 
-    pip install --upgrade MDAnalysis[analysis]
+    pip install --upgrade MDAnalysis[analysis,extra_formats,parallel]
 
-To install/upgrade tests:
+To install or upgrade the :ref:`test suite <mdanalysistests>` (optional, ~90 MB):
 
 .. code-block:: bash
 
     pip install --upgrade MDAnalysisTests
 
-If you intend to use MDAnalysis in JupyterLab, you will have to install
-an extra package for the progress bar in analysis classes:
+If you're using MDAnalysis in **JupyterLab**, install `ipywidgets`_ for progress bar support:
 
 .. code-block:: bash
 
-    pip install nodejs
-    jupyter labextension install @jupyter-widgets/jupyterlab-manager
+    pip install ipywidgets
 
+.. _source-installation:
 
-Development versions
-====================
-To install development versions of MDAnalysis, you can compile it from source. In order to install from source, you will need ``numpy`` and ``cython``. See :ref:`create-virtual-environment` for instructions on how to create a full development environment.
+Install from source
+===================
+
+This section is for **users** who want to install a **specific stable version** of MDAnalysis
+(e.g., for reproducibility or to debug a release). This is **not** intended for development or code contributions.
+If you're a developer, see :ref:`development-installation`.
+
+To install from source, you must first ensure that your environment already contains all necessary dependencies.
+We recommend you check the :ref:`create-virtual-environment` to set up a clean development environment
+with all required dependencies. We recommend using `mamba`_ or `conda`_ to manage this setup.
+
+Follow the steps in:
+
+- `Set up with conda-forge`_
+- `Set up with mamba`_
+
+MDAnalysis uses different Git branches for different purposes:
+
+- **package-X.Y.Z**: These branches contain specific stable releases (e.g., ``package-2.9.0``) and are intended for users who need reproducibility, debugging, or version locking. You can browse them on the GitHub `Branches page`_ or view official releases on the `Releases page`_.
+- **develop**: The `develop`_ branch is the default and active development branch. If you're contributing to MDAnalysis or want the latest features before they're released, base your work on this branch (see :ref:`development-installation`).
+
+To install the **latest stable release** from source:
+
+.. code-block:: bash
+
+    git clone https://github.com/MDAnalysis/mdanalysis
+    cd mdanalysis
+    git checkout release-<latest-version>
```

Comment:
Is `git switch` preferred over `git checkout` nowadays?

According to https://stackoverflow.com/a/70454786 simple checkout and switch are equivalent (but switch is clearer):
```suggestion
    git switch release-<latest-version>
```